runtime/compiler/runtime/J9Profiler.cpp

```diff
@@ -2400,14 +2402,18 @@ TR_CallSiteInfo::computeEffectiveCallerIndex(TR::Compilation *comp, TR::list<std
       auto itr = callStack.begin(), end = callStack.end();
       while (cursor && itr != end)
          {
-         if (comp->fe()->getInlinedCallSiteMethod(cursor) != itr->first)
+         TR_OpaqueMethodBlock *method = itr->first;
+         itr++;
+         if (comp->fe()->getInlinedCallSiteMethod(cursor) != method || (cursor->_byteCodeInfo.getByteCodeIndex() != itr->second.getByteCodeIndex()))
```

Comment:
It's possible that `itr == end` here since we've already done `itr++`, so we need to make sure that `itr != end` before we dereference. I think though that if we checked here, it would make the check in the while condition almost but not quite redundant. Maybe something like this?
```
if (itr != end)
   {
   auto next = itr;
   next++;
   while (cursor && next != end)
      {
      // Use method from itr, bytecode index from next.
      // ... Check stuff, advance cursor ...
      itr++;
      next++;
      }
   }
```

Reply:
Changed the implementation in https://github.com/eclipse-openj9/openj9/compare/2525dfcb3062082ccba873bb8bf463c296e53c02..9d0222669b96b186e0b5b1f42208416a2d9d371d and in https://github.com/eclipse-openj9/openj9/pull/16205/commits/fa08da05a11513ce329a00a98f1d60fe248519fc

Will squash them once you are OK with changes.